### PR TITLE
feat: MicrobatchRule full functionality and MicrobatchFlow model

### PR DESCRIPTION
**Pull request by riddick-oe**

### CODE REVIEW

File: src/microbatching/views/microbatch_flow.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    "history": {
+        "model_name": "history",
+        "related_name": "history",
+        "headers": ["ID", "History Date", "History Type", "History User"],
+        "fields": ["history_id", "history_date", "history_type", "history_user"],
+        "show_edit_actions": False,
+    },
+}
+
+# MICROBATCH_FLOW_STEP_FORMSET_FORM_FIELDS = [
```
Comment (loststoner) [inline src/microbatching/views/microbatch_flow.py:45]: Remove the comments if they are not necessary.

File: src/microbatching/utils.py
Comment (loststoner) [inline src/microbatching/utils.py:1]: If I'm not mistaken, the `get_related_fields`, `get_models_fields`, and `check_criteria_match` functions are the same. Can we import them from the `resource_assigner` app and reuse them here?

File: src/microbatching/models/microbatch_flow.py
Comment (loststoner) [inline src/microbatching/models/microbatch_flow.py:1]: The `description` field should have `null=True` and `blank=True`. Can we make the `max_flow_length` field optional and add a default value to the `min_flow_length` field, if possible?

Comment (riddick-oe) [reply src/microbatching/models/microbatch_flow.py:1]: I've added default values to both fields.

File: src/microbatching/utils.py
Comment (riddick-oe) [reply src/microbatching/utils.py:1]: I have moved these to the common app, as multiple apps are using them.

**Review by loststoner — APPROVED** [review]

**Pull request merged**